src/components/Navbar.tsx

```diff
@@ -17,24 +21,235 @@ const ConnectButton = styled(Button)`
+                    />
+                    <ReactModal
+                        isOpen={modalIsOpen}
+                        contentLabel="Connect a wallet"
+                        style={customStyles}
+                    >
+                        <StyledModalContent>
+                            <ModalHeader>
+                                <h2>Account</h2>
+                                <CloseButton onClick={closeModal}>
+                                    <svg
+                                        width="16"
+                                        height="16"
+                                        viewBox="0 0 16 16"
+                                        fill="none"
+                                        xmlns="http://www.w3.org/2000/svg"
+                                    >
+                                        <path
+                                            d="M15 1L0.999999 15"
+                                            stroke="#59799C"
+                                            stroke-width="1.5"
+                                            stroke-linecap="round"
+                                        />
+                                        <path
+                                            d="M1 1L15 15"
+                                            stroke="#59799C"
+                                            stroke-width="1.5"
+                                            stroke-linecap="round"
+                                        />
+                                    </svg>
+                                </CloseButton>
+                            </ModalHeader>
+                            <ContentContainer>
+                                <ContentHeader>
+                                    <Subheading>
+                                        Connected with Metamask
+                                    </Subheading>
+                                    <ChangeButton>Change</ChangeButton>
+                                </ContentHeader>
+                                <input
+                                    disabled={true}
+                                    placeholder={trunc(account)}
+                                />
+                                <LinkContainer>
+                                    <ExternalLink
+                                        href={`https://etherscan.io/address/${account}`}
+                                        target="_blank"
+                                        rel="noreferrer"
+                                    >
+                                        <ExternalLinkIcon /> View on explorer
+                                    </ExternalLink>
+                                    <ExternalLink
+                                        onClick={() => {
+                                            navigator.clipboard.writeText(
+                                                account
+                                            )
```

Comment:
Can this also change the text to "Copied!" after a click. Ideally I'd change it back to "Copy Address" after 2-3s. You can play around or just leave it at "Copied!" `#nice2have`